### DISCUSSION TITLE
Tailwind CSS Custom Typography

### DISCUSSION THREAD

**Discussion by sandranrivas** [tailwindlabs/tailwindcss#8232] [Help]
I'm trying to create a custom typography for my app.

This are my classes. as an example:

**main.css**

    :root {
          --typescale-display-large:
            font-size: 57px,
            line-height: 64;
        
          --typescale-display-medium:
            font-size: 45px,
            line-height: 52;
        
          --typescale-display-small:
            font-size: 35px,
            line-height: 50;
    }

after downloading 

npm install -D @tailwindcss/typography

**in tailwind.config.js** inside in extend.

      typography: {
        'display-large': 'var(--typescale-display-large)',
        'display-medium': 'var(--typescale-display-medium)',
        'display-small': 'var(--typescale-display-small)',
      },

I just try it this way, but I didn't find an example in the documentation of using typography for writting custom typography, what I read, this was just for overriding  the typography classes in tailwind. and it still doesn't work.

When trying to use this class. I tried using:

       <p class="typography-display-large">hi</p>

I also tried 

     <p class="prose prose-display-large">hi</p>

But it doesn't work. 
here is a link where from the docu I read https://tailwindcss.com/docs/typography-plugin

**Comment by DanielOberlechner** [comment]
Have you done it like this:
module.exports = {
  theme: {
    extend: {
      typography: ({ theme }) => ({
        pink: {
          css: {
            '--tw-prose-body': theme('colors.pink[800]'),
            '--tw-prose-headings': theme('colors.pink[900]'),
            '--tw-prose-lead': theme('colors.pink[700]'),
            '--tw-prose-links': theme('colors.pink[900]'),
            '--tw-prose-bold': theme('colors.pink[900]'),
            '--tw-prose-counters': theme('colors.pink[600]'),
            '--tw-prose-bullets': theme('colors.pink[400]'),
            '--tw-prose-hr': theme('colors.pink[300]'),
            '--tw-prose-quotes': theme('colors.pink[900]'),
            '--tw-prose-quote-borders': theme('colors.pink[300]'),
            '--tw-prose-captions': theme('colors.pink[700]'),
            '--tw-prose-code': theme('colors.pink[900]'),
            '--tw-prose-pre-code': theme('colors.pink[100]'),
            '--tw-prose-pre-bg': theme('colors.pink[900]'),
            '--tw-prose-th-borders': theme('colors.pink[300]'),
            '--tw-prose-td-borders': theme('colors.pink[200]'),
            '--tw-prose-invert-body': theme('colors.pink[200]'),
            '--tw-prose-invert-headings': theme('colors.white'),
            '--tw-prose-invert-lead': theme('colors.pink[300]'),
            '--tw-prose-invert-links': theme('colors.white'),
            '--tw-prose-invert-bold': theme('colors.white'),
            '--tw-prose-invert-counters': theme('colors.pink[400]'),
            '--tw-prose-invert-bullets': theme('colors.pink[600]'),
            '--tw-prose-invert-hr': theme('colors.pink[700]'),
            '--tw-prose-invert-quotes': theme('colors.pink[100]'),
            '--tw-prose-invert-quote-borders': theme('colors.pink[700]'),
            '--tw-prose-invert-captions': theme('colors.pink[400]'),
            '--tw-prose-invert-code': theme('colors.white'),
            '--tw-prose-invert-pre-code': theme('colors.pink[300]'),
            '--tw-prose-invert-pre-bg': 'rgb(0 0 0 / 50%)',
            '--tw-prose-invert-th-borders': theme('colors.pink[600]'),
            '--tw-prose-invert-td-borders': theme('colors.pink[700]'),
          },
        },
      }),
    },
  },
  plugins: [
    require('@tailwindcss/typography'),
    // ...
  ],
}

Look at the end here, at plugins require and then tailwindcsss typography have you put this into your tailwind.config.js ?

Kind regards,
Daniel Oberlechner

**Reply by sandranrivas** [reply]
hey thanks. But my custom Typography doesn't work. Just the predefined Tailwind classes. ex. prose, prose-2xl.

**Reply by DanielOberlechner** [reply]
@sandranrivas yeah I do understand that :)
But if you want it to work you have to install tailwindcss/typography!

You can read more about it right here:
https://tailwindcss.com/docs/typography-plugin

First I wasn't aware either that I need a plugin for this, but it is what it is ....

Kind regards,
Daniel Oberlechner :)

**Reply by sandranrivas** [reply]
Yeah, so I already installed the Plugin, and added it to the plugins. that was what I meant in my comment before. but my classes are still not working.
```

   'display-large': 'var(--typescale-display-large)',
   'display-medium':'var(--typescale-display-medium)',
   'display-small': 'var(--typescale-display-small)',
```
So my question was If I could use my custom classes for the Typography.
 The ones from tailwind do work. 
'prose' works.

But for example when trying to use 'display-large' as a class it does not work.

**Reply by davidstrouk** [reply]
Hey @sandranrivas, did you figure it out eventually? I'm now encountering the same issue. Thanks

**Reply by wongjn** [reply]
1. Rework the CSS that has the variables to valid CSS, like:
   ```css
   :root {
     --typescale-display-large-font-size: 57px;
     --typescale-display-large-line-height: 64px;
     --typescale-display-medium-font-size: 45px;
     --typescale-display-medium-line-height: 52px;
     --typescale-display-small-font-size: 35px;
     --typescale-display-small-line-height: 50px;
   }
   ```
2. Add the corresponding values to `typography` key in the Tailwind configuration:
   ```js
   typography: {
     'display-large': {
       css: {
         fontSize: 'var(--typescale-display-large-font-size)',
         lineHeight: 'var(--typescale-display-large-line-height)',
       }
     },
     'display-medium': {
       css: {
         fontSize: 'var(--typescale-display-medium-font-size)',
         lineHeight: 'var(--typescale-display-medium-line-height)',
       }
     },
     'display-small': {
       css: {
         fontSize: 'var(--typescale-display-small-font-size)',
         lineHeight: 'var(--typescale-display-small-line-height)',
       }
     },
   }
   ```
   Note that you may want to adjust the `fontSize`/`lineHeight` of other elements, like `h1`-`6` tags, etc.
3. Use the new classes:
   ```html
   <div class="prose prose-display-large">
     …
   </div>
   <div class="prose prose-display-medium">
     …
   </div>
   <div class="prose prose-display-small">
     …
   </div>
   ```

[Full working example Tailwind Play](https://play.tailwindcss.com/vQaOaahsgi)
